### PR TITLE
Stepped process: Adding dark variant

### PR DESCRIPTION
**Pull request by louismaximepiton**

### CODE REVIEW

**Comment by isabellechanclou** [comment]
@julien-deramond: design specifications for states (hover and pressed) are lacking and will be asked to designers.

**Comment by julien-deramond** [comment]
Haven't looked at the code but would it be possible for the "arrow separators" to be transparent rather than black (or white by default)? It would allow to use the dark variant (or the light one) on whatever background color.
Screenshot from 2022-08-24 14-37-22

**Comment by louismaximepiton** [comment]
> Haven't looked at the code but would it be possible for the "arrow separators" to be transparent rather than black (or white by default)? It would allow to use the dark variant (or the light one) on whatever background color.
> 
> Here is an example of rendering with a `.bg-warning`: Screenshot from 2022-08-24 14-37-22

I don't think so, or at least I can't see any way atm. To sum up the behavior, the arrow shape comes from a `::after` with `background-clip` and the small border comes from a `drop-shadow` implemented in the next item. If the `drop-shadow`is removed, set as transparent or inherit, we won't have the small border anymore. However, if you have any idea, feel free to make the changes on the PR directly :)

**Comment by julien-deramond** [comment]
> I don't think so, or at least I can't see any way atm. To sum up the behavior, the arrow shape comes from a `::after` with `background-clip` and `background-color: inherit` and the small border comes from a `drop-shadow` implemented in the next item. If the `drop-shadow`is removed, set as transparent or inherit, we won't have the small border anymore. However, if you have any idea, feel free to make the changes on the PR directly smile

Indeed the way it is developed (and maybe there's no other way to do that) doesn't allow any transparency. So let's let it as it is for now.

File: scss/_variables.scss [hunk 1]
```diff
@@ -1989,10 +1989,19 @@ $step-item-arrow-shape:       polygon(0% 0%, 1px 0%, subtract(100%, $border-widt
...
+$step-link-next-color:        $black !default;
 $step-link-line-height:       $line-height-sm !default;
 $step-link-marker:            counter(var(--bs-stepped-process-counter)) inspect("\A0") !default;
 $step-link-marker-lg:         counter(var(--bs-stepped-process-counter)) inspect(".\A0") !default;
 $step-link-text-decoration:   $link-decoration !default;
+
+$step-item-dark-bg:           $white !default;
+$step-item-dark-active-bg:    $brand-orange !default;
+$step-item-dark-next-bg:      $gray-700 !default;
+$step-item-dark-drop-shadow:  drop-shadow($step-item-shadow-size 0 0 $black) #{"/* rtl:"} drop-shadow(-$step-item-shadow-size 0 0 $black) #{"*/"} !default;
```
Comment (julien-deramond) [inline scss/_variables.scss:2001]: Don't remember if the fix was here, but there is an issue with the black "arrow" and its edges:

Screenshot from 2022-08-30 11-04-26

**Review by julien-deramond — CHANGES_REQUESTED** [review]

File: site/content/docs/5.2/components/stepped-process.md [hunk 2]
```diff
@@ -39,6 +39,33 @@ Add `.active` to a `.stepped-process-item` to indicate the current step, alongsi
 </nav>
 {{< /example >}}
 
+## Dark variant
```
[the remark this answers is not in the document]
Comment (louismaximepiton) [reply site/content/docs/5.2/components/stepped-process.md:42]: However, it is different in [Close buttons](https://twbs-bootstrap.netlify.app/docs/5.2/components/close-button/#white-variant), [Dropdowns](https://twbs-bootstrap.netlify.app/docs/5.2/components/dropdowns/#dark-dropdowns) maybe we did it in the wrong way in accordions ?

[the remark this answers is not in the document]
Comment (julien-deramond) [reply site/content/docs/5.2/components/stepped-process.md:42]: So let's try to be consistent with what's in Bootstrap (if it's consistent there).

File: scss/_variables.scss (hunk 1 above)
Comment (julien-deramond) [reply scss/_variables.scss:2001]: Seems to be a cache issue

**Review by julien-deramond — APPROVED** [review]
LGTM! Let's merge and come back to it if we have new information from the designers regarding some missing states.

**Pull request merged**